pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,233 @@
+        the index of the time_slice that the node belongs to.
+
+        Edges are represented as links between the nodes.
+
+        Parameters:
+        ----------
+        data: Data to initialize graph.  If data=None (default) an empty
+              graph is created.  The data can be an edge list, or any
+              NetworkX graph object
+
+        Examples:
+        --------
+        Create an empty Dynamic Bayesian Network with no nodes and no edges
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+
+        adding nodes and edges inside the dynamic bayesian network. A single
+        node can be added using the method below.
+
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('G',0),('L',0))])
+
+        Most of the methods will be imported from Bayesian Model
+
+        >>> dbn.nodes()
+        ['L', 'G', 'S', 'I', 'D']
+        >>> dbn.edges()
+        [(('D', 0), ('G', 0)), (('G', 0), ('L', 0)), (('I', 0), ('G', 0))]
+
+        If some edges connect nodes not yet in the model, the nodes
+        are added automatically. There are no errors when adding
+        nodes or edges that already exist.
+
+        Methods:
+        -------
+        add_nodes_from
+        add_edge
+        add_edges_from
+        intra_slice
+        inter_slice
+        add_cpds
+        initialize_initial_state
+        """
+        super().__init__()
+        if ebunch:
+            self.add_nodes_from(ebunch)
+
+    def add_edge(self, start, end, **kwargs):
+        """
+        Add an edge between two nodes.
+
+        The nodes will be automatically added if they are
+        not already in the initial static Bayesian Network.
+
+        Parameters
+        ----------
+        start, end: The start, end nodes should contain the
+                    (node_name, time_slice)
+                    Here, node_name can be a hashable python object
+                    while the time_slice is an integer value, which
+                    denotes the index of the time_slice that the node
+                    belongs to.
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['D', 'I'])
+        >>> dbn.add_edge(('D',0), ('I',0))
+        >>> dbn.edges()
+        [(('D', 0), ('I', 0))]
+        """
+        if not (isinstance(start, (tuple, list)) and isinstance(end, (tuple, list))): 
+            raise ValueError('the nodes inside the edge must be enclosed in a list or a tuple')
+
+        if not (start[1] in (0, 1) and end[1] in (0, 1)):
+            raise ValueError('the timeslices inside the node must belong to 0 or 1')
+
+        if start[1] == end[1]:
+            super().add_edge((start[0], 1 - start[1]), (end[0], 1 - end[1]))
+
+        super().add_edge(start, end)
+        self.correct_nodes()
+
+    def correct_nodes(self):
```

Comment:
If this method is only used in ```add_edge()``` method, then I would suggest to not expose to the user. Rename it to ```_correct_nodes()``` method.